Ladybird/AppKit/UI/Tab.swift

```diff
@@ -0,0 +1,307 @@
+/*
+ * Copyright (c) 2023-2024, Tim Flynn <trflynn89@serenityos.org>
+ * Copyright (c) 2024, Douwe Zumker <douwezumker@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+import Foundation
+import AppKit
+import Ladybird.Utilities
+import LibCore
+import LibGfx
+import LibURL
+import LibWebView
+import SwiftUI
+public class Tab: NSWindow,LadybirdWebViewObserver {
+
+    private let WINDOW_WIDTH: CGFloat = 1000
+    private let WINDOW_HEIGHT: CGFloat = 800
+
+    var title: String {
+        get {
+            self.attributedTitle?.string ?? ""
+        }
+        set {
+            self.updateTabTitleAndFavicon()
+        }
+    }
+
+    var favicon: NSImage?
+    var searchPanel: SearchPanel
+    var inspectorController: InspectorController?
+
+    static var defaultFavicon: NSImage {
+        struct Static {
+            static let defaultFavicon: NSImage = {
+                let defaultFaviconPath = try! Core.Resource.load_from_uri("resource://icons/48x48/app-browser.png" as StaticString)
+                let nsDefaultFaviconPath = Ladybird.string_to_ns_string(defaultFaviconPath.filesystem_path()!)
```

Comment:
We should be able to re-use my extension to Swift.String from https://github.com/LadybirdBrowser/ladybird/pull/1175 to convert AK::String to  Swift.Strings.